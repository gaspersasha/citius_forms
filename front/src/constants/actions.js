export const USER_LOGIN = 'USER_LOGIN';
export const USER_LOGOUT = 'USER_LOGOUT';
export const USER_SET_STATE = 'USER_SET_STATE';

export const DETAILS_GET = 'DETAILS_GET';
export const DETAILS_POST = 'DETAILS_POST';
export const DETAILS_SET_STATE = 'DETAILS_SET_STATE';

export const TOGGLE_ORDER_SUMMARY = 'TOGGLE_ORDER_SUMMARY';

export const CONTACT_DEALER_POST = 'CONTACT_DEALER_POST';
export const CONTACT_DEALER_SET_STATE = 'CONTACT_DEALER_SET_STATE';
export const CONTACT_DEALER_GET_VEHICLE = 'CONTACT_DEALER_GET_VEHICLE';
