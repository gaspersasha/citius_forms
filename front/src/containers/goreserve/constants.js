export const PAYMENT_METHOD = {
  stripe: 'open-stripe-form',
  payPal: 'open-paypal',
};

export const PAYMENT_METHODS_OPTIONS = [
  {
    value: 'cash',
    option: 'I will buy outright',
  },
  {
    value: 'finance',
    option: 'I will buy on finance',
  },
];
