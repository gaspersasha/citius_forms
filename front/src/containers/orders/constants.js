// The note: type can be null
export const PAYMENT_TYPE = {
  DEBIT: 'debit',
  CREDIT: 'credit',
  PAID: 'paid',
};

export const ORDER_SECTION_TYPE = {
  financialSummary: {
    title: 'Financial summary',
    subtitle: 'Deposit payments',
  },
  yourOrders: {
    title: 'Your orders',
    subtitle: 'Orders',
  },
};
