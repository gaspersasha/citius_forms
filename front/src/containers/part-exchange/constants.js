export const CAR_CONDITION_OPTIONS = [
  { value: 'great', option: 'Great' },
  { value: 'good', option: 'Good' },
  { value: 'fair', option: 'Fair' },
  { value: 'poor', option: 'Poor' },
];

export const SERVICE_HISTORY_OPTIONS = [
  { value: 'full', option: 'Full' },
  { value: 'part', option: 'Part' },
  { value: 'notdue', option: 'Not due' },
  { value: 'none', option: 'None' },
];
