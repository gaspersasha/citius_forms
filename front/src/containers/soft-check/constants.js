export const KEYS = [
  'title',
  'firstName',
  'middleName',
  'lastName',
  'email',
  'phone',
  'dateOfBirth',
  'postCode',
  'houseName',
  'houseNumber',
  'street',
  'town',
  'county',
  'currentIncome',
  'district',
];
export const MAX_SALARY = 9999999999.99;
export const SEARCH_PRICE_TAGS_LIST = [100, 150, 200, 300, 500];
