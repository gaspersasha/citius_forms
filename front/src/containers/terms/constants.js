export const MILAGE_OPTIONS = [
  { value: 6000, option: '6,000 miles' },
  { value: 7000, option: '7,000 miles' },
  { value: 8000, option: '8,000 miles' },
  { value: 9000, option: '9,000 miles' },
  { value: 10000, option: '10,000 miles' },
  { value: 11000, option: '11,000 miles' },
  { value: 12000, option: '12,000 miles' },
  { value: 13000, option: '13,000 miles' },
  { value: 14000, option: '14,000 miles' },
  { value: 15000, option: '15,000 miles' },
  { value: 16000, option: '16,000 miles' },
  { value: 17000, option: '17,000 miles' },
  { value: 18000, option: '18,000 miles' },
  { value: 19000, option: '19,000 miles' },
  { value: 20000, option: '20,000 miles' },
  { value: 21000, option: '21,000 miles' },
  { value: 22000, option: '22,000 miles' },
  { value: 23000, option: '23,000 miles' },
  { value: 24000, option: '24,000 miles' },
  { value: 25000, option: '25,000 miles' },
  { value: 26000, option: '26,000 miles' },
  { value: 27000, option: '27,000 miles' },
  { value: 28000, option: '28,000 miles' },
  { value: 29000, option: '29,000 miles' },
  { value: 30000, option: '30,000 miles' },
];

export const MONTHS_OPTIONS = [
  { value: 24, option: '24 months' },
  { value: 36, option: '36 months' },
  { value: 48, option: '48 months' },
  { value: 60, option: '60 months' },
];

export const FINANCE_OPTIONS = [
  { value: 'PCP', option: 'PCP' },
  { value: 'HP', option: 'HP' },
];

export const DELIVERY_INFO = [
  {
    value: 18,
    option: 'Driven delivery within England and Wales 239.0',
  },
  {
    value: 19,
    option: 'Driven/Ferry delivery to Belfast 599.0',
  },
  {
    value: 20,
    option: 'Driven delivery to Scotland 355.0',
  },
  {
    value: 21,
    option: 'Transporter delivery within England and Wales 479.0',
  },
  {
    value: 22,
    option: 'Transporter delivery to Scotland 599.0 ',
  },
];

export const FINANCE_CREDIT_RATING_OPTIONS = [
  { value: 'excellent', option: 'Excellent' },
  { value: 'good', option: 'Good' },
  { value: 'average', option: 'Average' },
  { value: 'poor', option: 'Poor' },
];

export const FINANCE_TYPE_CONTENT = {
  PCP: {
    title: '(PCP) Personal Contract Purchase',
    subtitle:
      'You make low monthly payments. At the end you either return the cars or buy it for an additional fee, often known as a baloon payment.',
    description:
      ' Personal Contract Purchase is a finance plan that allows you to defer part of the cost of the vehicle until the end of the agreement, reducing the regular monthly payments. The deferred element known as the guaranteed future value or optional final payment is based upon the likely value of the vehicle at the end of the agreement and is set by the finance company based upon the annual mileage selected. At the end of the term, you have the option to hand the vehicle back, part-exchange it or pay to final payment to take ownership of the vehicle. If you choose to hand the vehicle back excess mileage and damage charges may apply.',
    max: 40,
  },
  HP: {
    title: '(HP) Hire Purchase',
    subtitle:
      'You pay for the car in a series of installments. At the end of the agreed repayment period, it is yours.',
    description:
      'Hire Purchase is a finance plan that allows you to pay for the vehicle by fixed monthly payments over the term of the agreement. Hire purchase is a secured loan so the finance company has the vehicle as security for the loan until all payments and fees are made, at the end of the term of the agreement after all payments and option to purchase fee are made you take full ownership of the vehicle.',
    max: 80,
  },
};

export const NEW_FINANCE_QUOTE = {
  PCP: {
    capId: '49573',
    financeType: 'PCP',
    numRegularPayments: 46,
    paymentAmount: 152.16,
    apr: 8.9,
    finalPayment: 3173.34,
    gfv: 0,
    basicPrice: 9238.8,
    rateOfInterest: 4.61,
    flatRate: 0,
    totalCashPrice: 9238.8,
    firstPayment: 152.16,
    documentationFee: 0,
    purchaseFee: 0,
    totalPayable: 11248.86,
    totalCharges: 0,
    commission: 0,
    quoteReference:
      'https://services.codeweavers.net/forms/quote/5ba82157-a463-4754-b560-7385b88dd237',
    totalDeposit: 924,
    mileagePA: 8000,
    periodMonths: 36,
    amountFinanced: 8314.8,
    financeQuoteRequestDetails: 15,
    financeKey: 'ALPHERA_PCP',
  },
  HP: {
    capId: '49573',
    financeType: 'Hire Purchase',
    numRegularPayments: 58,
    paymentAmount: 170.8,
    apr: 8.9,
    finalPayment: 170.8,
    gfv: 0,
    basicPrice: 9238.8,
    rateOfInterest: 4.65,
    flatRate: 0,
    totalCashPrice: 9238.8,
    firstPayment: 170.8,
    documentationFee: 0,
    purchaseFee: 0,
    totalPayable: 11172,
    totalCharges: 0,
    commission: 0,
    quoteReference:
      'https://services.codeweavers.net/forms/quote/a7ccf346-ef3c-471f-b718-4a9c40606a99',
    totalDeposit: 924,
    mileagePA: 8000,
    periodMonths: 36,
    amountFinanced: 8314.8,
    financeQuoteRequestDetails: 26,
    financeKey: 'ALPHERA_HP',
  },
};
