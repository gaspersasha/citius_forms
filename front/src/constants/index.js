import * as ACTIONS from './actions';
import * as FORM from './form';
import * as URL from './urls';
import * as SCC from './scc';
import * as CONTACTS from './contacts';
import * as ENV from './env';

// TODO @ORDER_SUMMARY_DUMMY. Mock props
import * as ORDER_SUMMARY_DUMMY from './OrderSummaryDummy.json';

export { ACTIONS };
export { FORM };
export { URL };
export { CONTACTS };
export { ENV };
export { SCC };
export { ORDER_SUMMARY_DUMMY };
export { default as DEVICE_TYPES } from './device-types';
export { default as INPUT_STATUS } from './input-status';
export { default as VALIDATION_TYPES } from './validation-types';
