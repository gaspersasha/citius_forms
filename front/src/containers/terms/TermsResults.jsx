import React from 'react';
import PropTypes from 'prop-types';

import { TapButtons } from '~components';
import { IntlGBP } from '~utils';
import { FINANCE_OPTIONS, FINANCE_TYPE_CONTENT } from './constants';

import s from './terms.module.sass';

const TermsResults = ({
  quotes,
  disabledForm,
  handleChange,
  showModal,
  fields,
  apiError,
}) => (
  <>
    {Object.keys(quotes).length === 0 || apiError ? (
      <div>
        Sorry, we canâ€™t display a finance quote for your selected options,
        please complete a finance application and we will get back to your with
        an accurate quote.
      </div>
    ) : (
      Object.values(quotes).map((quote) => {
        const optionsIndex = quote.financeType === 'PCP' ? 0 : 1;

        return (
          <div
            key={quote.financeType}
            className={s.radioButtons}
            disabled={disabledForm(quote.financeType)}
            onClick={handleChange}
            id={quote.financeType}
            role="button"
          >
            <div className={s.info} id={quote.financeType}>
              <TapButtons
                options={[FINANCE_OPTIONS[optionsIndex]]}
                value={fields.finance.value}
                status={fields.finance.status}
                onChange={handleChange}
                label=""
              />
              <button
                className={s.infoModalBtn}
                onClick={() => showModal(quote.financeType)}
                type="button"
              >
                What is&nbsp;
                {quote.financeType}?
              </button>
            </div>
            <div className={s.InfoMonthPayments} id={quote.financeType}>
              <p id={quote.financeType}>
                {quote.periodMonths !== '' ? quote.periodMonths : 0}
                &nbsp; monthly payments of
              </p>
              <h2 id={quote.financeType}>
                {FINANCE_TYPE_CONTENT.PCP.max &&
                  IntlGBP.format(quote.firstPayment)}
              </h2>
            </div>
            <hr id={quote.financeType} />
            {quote.financeType === 'PCP' ? (
              <div id={quote.financeType}>
                <p className={s.info} id={quote.financeType}>
                  APR &nbsp;
                  {quote.apr}%
                </p>
                <div className={s.info} id={quote.financeType}>
                  <p id={quote.financeType}>Optional final payment</p>
                  <p id={quote.financeType}>
                    {IntlGBP.format(quote.finalPayment)}
                  </p>
                </div>
              </div>
            ) : (
              <p className={s.info} id={quote.financeType}>
                APR &nbsp;
                {quote.apr}%
              </p>
            )}
          </div>
        );
      })
    )}
  </>
);

TermsResults.propTypes = {
  handleChange: PropTypes.func.isRequired,
  disabledForm: PropTypes.func.isRequired,
  quotes: PropTypes.object.isRequired,
  fields: PropTypes.any.isRequired,
  showModal: PropTypes.func.isRequired,
  apiError: PropTypes.string.isRequired,
};

export default TermsResults;
