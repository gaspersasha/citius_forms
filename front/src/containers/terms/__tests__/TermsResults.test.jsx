import React from 'react';
import { INPUT_STATUS, VALIDATION_TYPES } from '~constants';
import TermsResults from '../TermsResults';
import { NEW_FINANCE_QUOTE } from '../constants';

const props = {
  fields: {
    deposit: {
      value: '',
      status: INPUT_STATUS.DEFAULT,
      validationType: VALIDATION_TYPES.NOT_EMPTY,
    },
    mileage: {
      value: '',
      status: INPUT_STATUS.DEFAULT,
      validationType: VALIDATION_TYPES.NOT_EMPTY,
    },
    terms: {
      value: '',
      status: INPUT_STATUS.DEFAULT,
      validationType: VALIDATION_TYPES.NOT_EMPTY,
    },
    finance: {
      value: '',
      status: INPUT_STATUS.DEFAULT,
      validationType: VALIDATION_TYPES.NOT_EMPTY,
    },
    delivery: {
      value: '',
      status: INPUT_STATUS.DEFAULT,
      validationType: VALIDATION_TYPES.NOT_EMPTY,
    },
    credit: {
      value: '',
      status: INPUT_STATUS.DEFAULT,
      validationType: VALIDATION_TYPES.NOT_EMPTY,
    },
  },
  disabledForm: jest.fn(),
  handleChange: jest.fn(),
  showModal: jest.fn(),
  quotes: { ...NEW_FINANCE_QUOTE },
  apiError: '',
};

describe('<TermsResults />', () => {
  let wrapper;
  const getWrapper = (method = shallow) => method(<TermsResults {...props} />);

  it('rendered without errors', () => {
    wrapper = getWrapper();

    expect(wrapper).toMatchSnapshot();
  });

  it('button onClick responds to changes', () => {
    wrapper = getWrapper(mount);
    const PCP = '[id="PCP"]';
    const HirePurchase = '[id="Hire Purchase"]';

    wrapper.find(PCP).at(1).simulate('click');
    wrapper.find(HirePurchase).at(1).simulate('click');
  });

  it('onClick', () => {
    const getChild = () => wrapper.find('button');

    getChild().at(1).invoke('onClick')('PCP');
    expect(props.showModal).toBeCalled();
  });
});
