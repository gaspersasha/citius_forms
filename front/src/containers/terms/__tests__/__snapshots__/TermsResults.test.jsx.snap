// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TermsResults /> rendered without errors 1`] = `
<Fragment>
  <div
    className="radioButtons"
    id="PCP"
    key="PCP"
    onClick={[MockFunction]}
    role="button"
  >
    <div
      className="info"
      id="PCP"
    >
      <TapButtons
        label=""
        name=""
        onChange={[MockFunction]}
        options={
          Array [
            Object {
              "option": "PCP",
              "value": "PCP",
            },
          ]
        }
        status="default"
        subTitle=""
        value=""
      />
      <button
        className="infoModalBtn"
        onClick={[Function]}
        type="button"
      >
        What is 
        PCP
        ?
      </button>
    </div>
    <div
      className="InfoMonthPayments"
      id="PCP"
    >
      <p
        id="PCP"
      >
        36
          monthly payments of
      </p>
      <h2
        id="PCP"
      >
        £152
      </h2>
    </div>
    <hr
      id="PCP"
    />
    <div
      id="PCP"
    >
      <p
        className="info"
        id="PCP"
      >
        APR  
        8.9
        %
      </p>
      <div
        className="info"
        id="PCP"
      >
        <p
          id="PCP"
        >
          Optional final payment
        </p>
        <p
          id="PCP"
        >
          £3,173
        </p>
      </div>
    </div>
  </div>
  <div
    className="radioButtons"
    id="Hire Purchase"
    key="Hire Purchase"
    onClick={[MockFunction]}
    role="button"
  >
    <div
      className="info"
      id="Hire Purchase"
    >
      <TapButtons
        label=""
        name=""
        onChange={[MockFunction]}
        options={
          Array [
            Object {
              "option": "HP",
              "value": "HP",
            },
          ]
        }
        status="default"
        subTitle=""
        value=""
      />
      <button
        className="infoModalBtn"
        onClick={[Function]}
        type="button"
      >
        What is 
        Hire Purchase
        ?
      </button>
    </div>
    <div
      className="InfoMonthPayments"
      id="Hire Purchase"
    >
      <p
        id="Hire Purchase"
      >
        36
          monthly payments of
      </p>
      <h2
        id="Hire Purchase"
      >
        £171
      </h2>
    </div>
    <hr
      id="Hire Purchase"
    />
    <p
      className="info"
      id="Hire Purchase"
    >
      APR  
      8.9
      %
    </p>
  </div>
</Fragment>
`;
